ranger-zookeeper/src/main/java/io/appform/ranger/zookeeper/servicefinder/ZkNodeDataSource.java

```diff
@@ -55,12 +56,12 @@ private List<ServiceNode<T>> checkForUpdateOnZookeeper(D deserializer) {
         if (!isStarted()) {
             log.warn("Data source is not yet started for service: {}. No nodes will be returned.",
                      service.getServiceName());
-            return Collections.emptyList();
+            return null;
         }
         if (isStopped()) {
             log.warn("Data source is  stopped already for service: {}. No nodes will be returned.",
                      service.getServiceName());
-            return Collections.emptyList();
+            return null;
```

Comment:
@koushikr @santanusinha Not sure about these two checks actually. Please advise.

Reply:
this is fine